browser/ui/tabs/split_view_browser_data.cc

```diff
@@ -27,7 +27,11 @@ SplitViewBrowserData::SplitViewBrowserData(Browser* browser)
       *this, browser->tab_strip_model());
 }
 
-SplitViewBrowserData::~SplitViewBrowserData() = default;
+SplitViewBrowserData::~SplitViewBrowserData() {
+  while (!observers_.empty()) {
+    observers_.begin()->OnWillDeleteBrowserData();
+  }
```

Comment:
This doesn't work?
```
  for (auto& observer : observers_) {
    observer. OnWillDeleteBrowserData();
  }
```

I think  observer list could handle removing while iterating.
```
//   A list of observers. Unlike a standard vector or list, this container can
//   be modified during iteration without invalidating the iterator. So, it
//   safely handles the case of an observer removing itself or other observers
//   from the list while observers are being notified.
```

Reply:
I think using for loop might be broken when observer deletes itself from |observers_|